library/sha256.c

```diff
@@ -44,6 +44,7 @@
 #      if defined(__linux__)
 /* Our preferred method of detection is getauxval() */
 #        include <sys/auxv.h>
+#        define HAVE_GETAUXVAL
```

Comment:
I do not think we need that. Auxiliary vectors are always available for linux. 
```suggestion
```

Reply:
But they might be available on other systems, hence `HAVE_GETAUXVAL`, which I don't think we should get rid of

Reply:
Yes. How about change it to `HAVE_AUXILIARY_VECTOR` . That is part of ELF auxiliary vertor.